data-prepper-plugins/failures-common/src/main/java/org/opensearch/dataprepper/plugins/dlq/s3/S3DlqWriter.java

```diff
@@ -0,0 +1,141 @@
+import org.opensearch.dataprepper.model.configuration.DataPrepperVersion;
+import org.opensearch.dataprepper.model.failures.DlqObject;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectResponse;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * S3 Dlq writer. Stores DLQ Objects in an S3 bucket.
+ *
+ * @since 2.2
+ */
+public class S3DlqWriter implements DlqWriter {
+
+    static final String S3_DLQ_RECORDS_SUCCESS = "dlqS3RecordsSuccess";
+    static final String S3_DLQ_RECORDS_FAILED = "dlqS3RecordsFailed";
+    static final String S3_DLQ_REQUEST_SUCCESS = "dlqS3RequestSuccess";
+    static final String S3_DLQ_REQUEST_FAILED = "dlqS3RequestFailed";
+    static final String S3_DLQ_REQUEST_LATENCY = "dlqS3RequestLatency";
+    static final String S3_DLQ_REQUEST_SIZE_BYTES = "dlqS3RequestSizeBytes";
+    private static final String KEY_NAME_FORMAT = "dlq-v%s-%s-%s-%s-%s";
+    private static final String FULL_KEY_FORMAT = "%s/%s";
+
+    private final S3Client s3Client;
+    private final String bucket;
+    private final String keyPathPrefix;
+    private final ObjectMapper objectMapper;
+
+    private final Counter dlqS3RecordsSuccessCounter;
+    private final Counter dlqS3RecordsFailedCounter;
+    private final Counter dlqS3RequestSuccessCounter;
+    private final Counter dlqS3RequestFailedCounter;
+    private final Timer dlqS3RequestTimer;
+    private final DistributionSummary dlqS3RequestSizeBytesSummary;
+
+    S3DlqWriter(final S3DlqWriterConfig s3DlqWriterConfig, final ObjectMapper objectMapper, final PluginMetrics pluginMetrics) {
+        dlqS3RecordsSuccessCounter = pluginMetrics.counter(S3_DLQ_RECORDS_SUCCESS);
+        dlqS3RecordsFailedCounter = pluginMetrics.counter(S3_DLQ_RECORDS_FAILED);
+        dlqS3RequestSuccessCounter = pluginMetrics.counter(S3_DLQ_REQUEST_SUCCESS);
+        dlqS3RequestFailedCounter = pluginMetrics.counter(S3_DLQ_REQUEST_FAILED);
+        dlqS3RequestTimer = pluginMetrics.timer(S3_DLQ_REQUEST_LATENCY);
+        dlqS3RequestSizeBytesSummary = pluginMetrics.summary(S3_DLQ_REQUEST_SIZE_BYTES);
+
+        this.s3Client = s3DlqWriterConfig.getS3Client();
+        Objects.requireNonNull(s3DlqWriterConfig.getBucket());
+        this.bucket = s3DlqWriterConfig.getBucket();
+        this.keyPathPrefix = s3DlqWriterConfig.getKeyPathPrefix();
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public void write(final List<DlqObject> dlqObjects, final String pipelineName, final String pluginId) throws IOException {
+        try {
+            doWrite(dlqObjects, pipelineName, pluginId);
+            dlqS3RequestSuccessCounter.increment();
```

Comment:
Do we need this ? We increment the counter based on dlq objects in next line.

Reply:
These are two different metrics. (ie Requests vs Records)